src/neo/SmartContract/Native/Designate/DesignateContract.cs

```diff
@@ -48,18 +80,22 @@ private void DesignateAsRole(ApplicationEngine engine, Role role, ECPoint[] node
             if (!Enum.IsDefined(typeof(Role), role))
                 throw new ArgumentOutOfRangeException(nameof(role));
             if (!CheckCommittee(engine)) throw new InvalidOperationException();
-            NodeList list = engine.Snapshot.Storages.GetAndChange(CreateStorageKey((byte)role)).GetInteroperable<NodeList>();
+            uint index = engine.Snapshot.Height + 1;
+            NodeList list = engine.Snapshot.Storages.GetAndChange(CreateStorageKey((byte)role).AddBigEndian(index), () => new StorageItem(new NodeList())).GetInteroperable<NodeList>();
```

Comment:
Maybe we should fault if we already set this role for this index

Reply:
It's not set role for index. it's we change node list at some index. 
New node list will take effect after the index and will not change until committee redesignate in the future.
I think there is little possibility committee send multiple designate transactions in one block. If they do, The last one will overwrite the previous.

Reply:
> The last one will overwrite the previous in the same block.

I want to avoid in the same block:
- CN: Change role
- CN: Submit TX
- CN: Restore role

I know that we should trust consensus nodes, but they shouldn't be able to change a role in the same block, this could only happen with bad intentions.

Reply:
Done.